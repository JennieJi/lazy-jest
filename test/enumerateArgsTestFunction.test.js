const { enumerateArgsTestFunction } = require('../src/testFunction');
const configArgs = require('../src/caseGenerator/configArgs');

describe('function with empty args', () => {
  const emptyArgsFunc = () => 'Success!';
  enumerateArgsTestFunction(emptyArgsFunc, configArgs());
});

describe('function has compulsory args only', () => {
  const myFunc = (a, b) => a + b;
  enumerateArgsTestFunction(
    myFunc,
    configArgs()
      .arg('a', [0, 1])
      .arg('b', [-1, 99999]),
    'no invalid case'
  );
  enumerateArgsTestFunction(
    myFunc,
    configArgs()
      .arg('a', [0, 1], { invalidCases: [NaN, null, ''] })
      .arg('b', [-1, 99999], { invalidCases: [NaN, null, ''] }),
    'has invalid cases'
  );
});

describe('function has optional args', () => {
  const myFunc = (a, b = 0) => a + b;
  enumerateArgsTestFunction(
    myFunc,
    configArgs()
      .arg('a', [0, 1], { invalidCases: [NaN, null, ''] })
      .arg('b', [-1, 99999], { invalidCases: [NaN, null, ''], optional: true })
  );
});
