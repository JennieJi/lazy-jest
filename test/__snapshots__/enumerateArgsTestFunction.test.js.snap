// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`function has compulsory args only has invalid cases - (["a", "b"]) - [0,-1] 1`] = `-1`;

exports[`function has compulsory args only has invalid cases - (["a", "b"]) - [0,99999] 1`] = `99999`;

exports[`function has compulsory args only has invalid cases - (["a", "b"]) - [1,-1] 1`] = `0`;

exports[`function has compulsory args only has invalid cases - (["a", "b"]) - [1,99999] 1`] = `100000`;

exports[`function has compulsory args only has invalid cases - (["a", "b"]) argument <a> is invalid - ["",-1] 1`] = `"-1"`;

exports[`function has compulsory args only has invalid cases - (["a", "b"]) argument <a> is invalid - [null,-1] 1`] = `NaN`;

exports[`function has compulsory args only has invalid cases - (["a", "b"]) argument <a> is invalid - [null,-1] 2`] = `-1`;

exports[`function has compulsory args only has invalid cases - (["a", "b"]) argument <b> is invalid - [0,""] 1`] = `"0"`;

exports[`function has compulsory args only has invalid cases - (["a", "b"]) argument <b> is invalid - [0,null] 1`] = `NaN`;

exports[`function has compulsory args only has invalid cases - (["a", "b"]) argument <b> is invalid - [0,null] 2`] = `0`;

exports[`function has compulsory args only no invalid case - (["a", "b"]) - [0,-1] 1`] = `-1`;

exports[`function has compulsory args only no invalid case - (["a", "b"]) - [0,99999] 1`] = `99999`;

exports[`function has compulsory args only no invalid case - (["a", "b"]) - [1,-1] 1`] = `0`;

exports[`function has compulsory args only no invalid case - (["a", "b"]) - [1,99999] 1`] = `100000`;

exports[`function has optional args myFunc() - (["a", "b"]) - [0,-1] 1`] = `-1`;

exports[`function has optional args myFunc() - (["a", "b"]) - [0,99999] 1`] = `99999`;

exports[`function has optional args myFunc() - (["a", "b"]) - [0,null] 1`] = `0`;

exports[`function has optional args myFunc() - (["a", "b"]) - [1,-1] 1`] = `0`;

exports[`function has optional args myFunc() - (["a", "b"]) - [1,99999] 1`] = `100000`;

exports[`function has optional args myFunc() - (["a", "b"]) - [1,null] 1`] = `1`;

exports[`function has optional args myFunc() - (["a", "b"]) argument <a> is invalid - ["",-1] 1`] = `"-1"`;

exports[`function has optional args myFunc() - (["a", "b"]) argument <a> is invalid - [null,-1] 1`] = `NaN`;

exports[`function has optional args myFunc() - (["a", "b"]) argument <a> is invalid - [null,-1] 2`] = `-1`;

exports[`function has optional args myFunc() - (["a", "b"]) argument <b> is invalid - [0,""] 1`] = `"0"`;

exports[`function has optional args myFunc() - (["a", "b"]) argument <b> is invalid - [0,null] 1`] = `NaN`;

exports[`function has optional args myFunc() - (["a", "b"]) argument <b> is invalid - [0,null] 2`] = `0`;

exports[`function has optional args myFunc() - (["a"]) - [0] 1`] = `0`;

exports[`function has optional args myFunc() - (["a"]) - [1] 1`] = `1`;

exports[`function has optional args myFunc() - (["a"]) argument <a> is invalid - [""] 1`] = `"0"`;

exports[`function has optional args myFunc() - (["a"]) argument <a> is invalid - [null] 1`] = `NaN`;

exports[`function has optional args myFunc() - (["a"]) argument <a> is invalid - [null] 2`] = `0`;

exports[`function with empty args emptyArgsFunc() - ([]) - () 1`] = `"Success!"`;
